crates/provider-sdk/src/lib.rs

```diff
@@ -7,25 +7,35 @@ use async_trait::async_trait;
 use serde::{Deserialize, Serialize};
 use tracing::{error, info, warn};
 
+use crate::error::{InvocationError, InvocationResult};
+
 pub mod error;
 pub mod provider;
-pub mod provider_main;
-pub mod rpc_client;
-
-pub use provider::ProviderConnection;
-pub use provider_main::{load_host_data, run_provider, start_provider};
-pub use rpc_client::RpcClient;
-
-pub use wasmcloud_tracing;
 
 /// Re-export of types from [`wasmcloud_core`]
 pub use core::{
     HealthCheckRequest, HealthCheckResponse, InterfaceLinkDefinition, WasmCloudEntity, WitFunction,
     WitInterface, WitNamespace, WitPackage,
 };
+pub use provider::*;
 pub use wasmcloud_core as core;
+pub use wasmcloud_tracing;
 
-use crate::error::{InvocationError, InvocationResult};
+pub fn link_put_subject(lattice: &str, provider_key: &str) -> String {
+    format!("wasmbus.rpc.{lattice}.{provider_key}.linkdefs.put")
+}
+
+pub fn link_del_subject(lattice: &str, provider_key: &str) -> String {
+    format!("wasmbus.rpc.{lattice}.{provider_key}.linkdefs.del")
+}
+
+pub fn health_subject(lattice: &str, provider_key: &str) -> String {
+    format!("wasmbus.rpc.{lattice}.{provider_key}.health")
+}
+
+pub fn shutdown_subject(lattice: &str, provider_key: &str, link_name: &str) -> String {
+    format!("wasmbus.rpc.{lattice}.{provider_key}.{link_name}.shutdown")
+}
```

Comment:
this should probably be in `core::rpc`

Reply:
I've moved it closer to usage, let's address this one later - I don't see similar functionality in core now